modules/context/context.go

```diff
@@ -340,6 +340,13 @@ func Contexter() macaron.Handler {
 		ctx.Data["EnableSwagger"] = setting.API.EnableSwagger
 		ctx.Data["EnableOpenIDSignIn"] = setting.Service.EnableOpenIDSignIn
 
+		if models.HasEngine {
```

Comment:
`EnableSSPI` is used only in `signin_navbar.tmpl` so no need to query database on each request but only on those routes that use it (probably should be moved directly to specific route handlers or moved to some other middleware that is used for these specific routes (most probably together with `EnableOpenIDSignIn`)

Reply:
Extracted the query to the DB to a separate function and moved the check in the route handler for the sign in page.
Left `EnableOpenIDSignIn` as it is, because it is just reading a settings variable and makes no DB queries.

Reply:
Should be fixed by commit [Only make a query to the DB to check if SSPI is enabled on handlers that need that information for templates](https://github.com/go-gitea/gitea/pull/8463/commits/616351c8b56cedf18a2e707d22a3c1952d56ef3e)